pkgs/top-level/static.nix

```diff
@@ -77,35 +77,251 @@ self: super: let
+
+  gtk3 = (super.gtk3.override {
+    trackerSupport = false;
+    withGtkDoc = false;
+    # Symbol overlap between wayland and libXcursor
+    waylandSupport = false;
+    wayland = null;
+    wayland-protocols = null;
+  }).overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.pkg-config
+      super.mesonShlibsToStaticHook
+      gdk-pixbuf
+    ];
+    mesonFlags = (old.mesonFlags or []) ++ [
+      "-Dintrospection=false"
+      "-Dwayland_backend=false"
+    ];
+    NIX_CFLAGS_COMPILE = "-DG_LOG_DOMAIN=\"\"\"\"";
+  });
+
+  harfbuzz = super.harfbuzz.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+  });
+
+  json-glib = super.json-glib.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+  });
+
+  libinput = super.libinput.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+  });
+
+  libselinux = (super.libselinux.override {
+    fts = super.musl-fts;
+    enablePython = false;
+  }).overrideAttrs (old: {
+    patches = (old.patches or []) ++ [
+      # Patch to disable shared libraries, a cleaner way
+      # would be to fork this to use meson which is much
+      # easier to tweak for static builds using NIX_MESON_DEPENDENCY_STATIC
+      # and/or the mesonShlibsToStaticHook
+      ../os-specific/linux/libselinux/fix-static-build.patch
+    ];
+  });
+
+  libsoup = super.libsoup.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+    NIX_LDFLAGS = (old.NIX_LDFLAGS or "") + " -lidn2 -lunistring";
+  });
+
+  libwacom = super.libwacom.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+  });
+
+  libxkbcommon = super.libxkbcommon.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+  });
+
   ocaml-ng = self.lib.mapAttrs (_: set:
     if set ? overrideScope' then set.overrideScope' ocamlStaticAdapter else set
   ) super.ocaml-ng;
 
+  p11-kit = super.p11-kit.overrideAttrs (old: {
+    postPatch = (old.postPatch or "") + ''
+      sed -i '/install_data(pkcs11_conf_example/,+1d' p11-kit/meson.build
+    '';
+  });
+
+  pango = super.pango.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+  });
+
   perl = super.perl.override {
     # Don’t use new stdenv zlib because
     # it doesn’t like the --disable-shared flag
     stdenv = super.stdenv;
   };
 
+  shared-mime-info = super.shared-mime-info.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
```

Comment:
maybe this should be the default and one can disable this if this was wrong, like dontAddStaticConfigureFlag, what do you think ?

Reply:
I will add this logic in the makeStaticLibraries adapter. BTW I don't  know why it's called like that because it's applied to both libraries and executables.